.circleci/config.yml

```diff
@@ -1,164 +1,113 @@
 version: 2.1
 
-parameters:
-  preview:
-    type: boolean
-    default: true
-  3-10:
-    type: boolean
-    default: true
-  3-11:
-    type: boolean
-    default: true
+# orbs:
+#   coveralls: coveralls/coveralls@2.2.1
+
+workflows:
+  ci:
+    jobs:
+      - lint
+      - test:
+          matrix:
+            parameters:
+              # TODO: Revisit why pyenv doesn't recognize 3.12
+              python_version: ["3.8", "3.9", "3.10", "3.11"] # "3.12"
+              arangodb_config: ["single", "cluster"]
+              arangodb_license: ["community", "enterprise"]
+              arangodb_version: ["3.10.10", "3.11.4", "latest"]
 
 jobs:
-  run-tests:
-    resource_class: small
+  lint:
+    docker:
+      - image: python:latest
+    steps:
+      - checkout
+      - run:
+          name: Install Dependencies
+          command: pip install .[dev]
+
+      - run:
+          name: Run black
+          command: black --check --verbose --diff --color --config=pyproject.toml ./arango ./tests/
+
+      - run:
+          name: Run flake8
+          command: flake8 ./arango ./tests
+
+      - run:
+          name: Run isort
+          command: isort --check ./arango ./tests
+
+      - run:
+          name: Run mypy
+          command: mypy ./arango
+
+  test:
     parameters:
-      python-version:
+      python_version:
         type: string
-        default: "latest"
-      arangodb-version:
+      arangodb_config:
         type: string
-        default: "arangodb:latest"
-      arangodb-config:
+      arangodb_license:
         type: string
-        default: "single.conf"
-      cluster:
-        type: boolean
-        default: false
-      enterprise:
-        type: boolean
-        default: false
-    docker:
-      - image: python:<< parameters.python-version >>
-        command: ["/bin/sh", "-c", "python -m http.server"]
-      - image: arangodb/<< parameters.arangodb-version >>
-        environment:
-          ARANGODB_CONF: << parameters.arangodb-config >>
-          PROJECT: /root/project
-        command:
-          - "/bin/sh"
-          - "-c"
-          - >
-            while ! wget -q -O /dev/null http://localhost:8000/$PROJECT/tests/static/setup.sh; do sleep 1; done &&
-            wget -O - http://localhost:8000/$PROJECT/tests/static/setup.sh |
-            /bin/sh
+      arangodb_version:
+        type: string
+    # TODO: Reconsider using a docker image instead of a machine
+    # i.e cimg/python:<< parameters.python_version >>
+    machine:
+      image: ubuntu-2204:current
```

Comment:
Hopefully we can revisit using `cimg/python` with `starter.sh` as that would cut the job times from ~4 minutes to ~2 minutes